main.go

```diff
@@ -48,7 +47,9 @@ func getGithubForks() error {
 		forks = append(forks, fork)
 
 	}
+	lock.Lock()
 	forksCache = forks
+	lock.Unlock()
```

Comment:
Because the call to `Unlock` is so close to `Lock`, this is not a problem, but there's a convention that people often use to ensure the `Unlock` call always happens. This matters more as the code logic becomes more complex. You'll see this convention a lot:

```suggestion
	lock.Lock()
	defer lock.Unlock()
	forksCache = forks
```

This ensures that, no matter what path the control flow happens to take at execution, the `Unlock` method will be called as the function is exiting. There are cases where this "defer unlock" convention doesn't work, but for the most part, it's a more reliable way to make sure your lock/unlock calls always happen.

Reply:
I have added the defer for the lock